server/android_notification_server.go

```diff
@@ -220,3 +213,31 @@ func (me *AndroidNotificationServer) SendNotification(msg *PushNotification) Pus
 	}
 	return NewOkPushResponse()
 }
+
+func isRetriableError(err error) bool {
+	// Any error that is not a known error, but the Internal error
+	return !(messaging.IsUnregistered(err) ||
+		messaging.IsInvalidArgument(err) ||
+		messaging.IsQuotaExceeded(err) ||
+		messaging.IsSenderIDMismatch(err) ||
+		messaging.IsThirdPartyAuthError(err))
+}
+
+func (me *AndroidNotificationServer) SendNotificationWithRetry(fcmMsg *messaging.Message, retry int) error {
+	ctx, cancel := context.WithTimeout(context.Background(), me.sendTimeout)
+	defer cancel()
+
+	start := time.Now()
+	_, err := me.client.Send(ctx, fcmMsg)
+	if me.metrics != nil {
+		me.metrics.observerNotificationResponse(PushNotifyAndroid, time.Since(start).Seconds())
```

Comment:
Should we add in the metrics the rety attempt? I don't see the counter being added to the metrics

Reply:
The counter would go after all the retries. No need to mark it as a failure if we are retrying.

We could add some different metric for that though. 0/5

Reply:
Yes I think we should add a metrics/ligs at least to have some idea that is indeed happening and why, that way it would allows us to narrow down what could be causing future double notifications etc...